### ISSUE TITLE
Site Editor: announce a summary of patterns upon entering Patterns section

### ISSUE THREAD

**Issue by annezazu** [WordPress/gutenberg#54632]
This came up as part of the [Hallway Hangout on improving accessibility in the Site Editor](https://make.wordpress.org/test/2023/09/14/hallway-hangout-lets-chat-about-improving-accessibility-in-the-site-editor/) when entering the Site Editor > Patterns section. This issue is discussed [starting at 25:15](https://youtu.be/6N3Au-7jLoE?si=1v-cAcECMAY-4cSY&t=1515) to around 26:15. Currently, it's not clear how many things you might eventually be looking at, especially if you don't have any patterns to begin with as the first thing announced is "my patterns zero".

**Comment by joedolson** [comment]
![image](https://github.com/WordPress/gutenberg/assets/1686451/eba1651a-f180-40cd-8af2-e58e49299786)

In the above screenshot, the first item a user will encounter is the indicator "My Patterns 0".  It's inside a parent div with `role="listitem"`, so it's semantically presented as a list with one item. This gives the impression that you have no patterns.

If this was integrated together with the list of standard patterns, so it was showed up in a list of several items, it would be clear that this is just one of a set of indicators. Alternately, if it was part of a list of comparable groups, e.g. 'My Patterns: 0', 'Theme Patterns: 19', 'Plugin Patterns: 423', that would give context for all relevant groups.

The groups above are purely hypothetical; but I think that there should be existing pattern groups that could be comparable to "My Patterns". One question: does "My Patterns" refer to "Patterns I, the current user, created" or is it "Patterns that were created for this site by any user"? If the latter, I don't think "My" is the appropriate ownership. That implies personal ownership, rather than site ownership.

**Comment by talldan** [comment]
It looks as though this has changed now, the default tab is 'All Patterns', which should hopefully give users a good overview of the total number of patterns:
![Screenshot 2023-10-09 at 12 27 27 pm](https://github.com/WordPress/gutenberg/assets/677833/24c0358a-7a4a-47f1-88a5-1156c8921d19)

Ok to close this?

**Comment by annezazu** [comment]
Hmm I don't think so considering the 35 number listed there doesn't match up to the total number of patterns. Curious to hear from others though.

**Comment by talldan** [comment]
> Hmm I don't think so considering the 35 number listed there doesn't match up to the total number of patterns. Curious to hear from others though.

In what way doesn't it match up?

To confirm, a single pattern can be in multiple categories, so the numbers shouldn't add up to form a total if that's what you mean.

**Comment by joedolson** [comment]
As long as the number represents how many patterns you'll find on opening that tab, I think it's fine. The change to start with 'All Patterns' resolves this issue for me.

**Comment by annezazu** [comment]
Awesome! The number don't add up when I look at the overall list of patterns below -- that threw me off :)
